### ISSUE TITLE
Feature request: More calc modes for merging

### ISSUE THREAD

**Issue by miasik** [ltdrdata/ComfyUI-Impact-Pack#146]
Please, port other calc modes for models merging.
Supermerger has really interesting cosA, cosB and amazing TrainDfference.

**Comment by ltdrdata** [comment]
Have you tried using this node?
https://github.com/WASasquatch/ComfyUI_Preset_Merger

**Comment by miasik** [comment]
> Have you tried using this node? https://github.com/WASasquatch/ComfyUI_Preset_Merger

I have not because I choose blocks and weights manually. Anyway the calcmethods are not about blocks and weights. They are more about math used to merge.
I'd like you to read a description about them:
https://github.com/hako-mikan/sd-webui-supermerger/blob/main/calcmode_en.md
and the starting discussion:
https://github.com/hako-mikan/sd-webui-supermerger/issues/33

**Comment by silveroxides** [comment]
Was this feature request ever revisited och considered? Asking because merge calcmode options have increased since initial post.
The current ones are "TrainDifference", "SmoothAdd", "Extract"(calculate and extract similarities/dissimilarities from model (B-A) or (C-A) and add resulting to model A), "Tensor weight average calculated weighted sum", "Plus Random"(no idea) and a repo called [sd-webui-untitledmerger](https://github.com/groinge/sd-webui-untitledmerger) has a calcmode specifically for add dissimilarities without subtraction of A as well as an implementation of DARE

**Comment by miasik** [comment]
> Was this feature request ever revisited och considered? Asking because merge calcmode options have increased since initial post. The current ones are "TrainDifference", "SmoothAdd", "Extract"(calculate and extract similarities/dissimilarities from model (B-A) or (C-A) and add resulting to model A), "Tensor weight average calculated weighted sum", "Plus Random"(no idea) and a repo called [sd-webui-untitledmerger](https://github.com/groinge/sd-webui-untitledmerger) has a calcmode specifically for add dissimilarities without subtraction of A as well as an implementation of DARE

No, I asked when available set was short, without the modes enlisted above by you.

**Comment by silveroxides** [comment]
> 
> No, I asked when available set was short, without the modes enlisted above by you.

It would seem that impact pack isn't the ideal target for those nodes since it does not contain any other nodes related to merging from what I can see. 

Most likely your best shot is either with WAS node suite repo and request there or see if the one running the comfyUI DARE nodes repo have interest in adding them to his repo.

**Comment by ltdrdata** [comment]
> > No, I asked when available set was short, without the modes enlisted above by you.
> 
> It would seem that impact pack isn't the ideal target for those nodes since it does not contain any other nodes related to merging from what I can see.
> 
> Most likely your best shot is either with WAS node suite repo and request there or see if the one running the comfyUI DARE nodes repo have interest in adding them to his repo.

I agree that implementing it from a repository composed of a project specialized in model merging would be better.
